model/integration_action.go

```diff
@@ -530,3 +532,136 @@ func DoPostActionRequestFromJson(data io.Reader) *DoPostActionRequest {
 	json.NewDecoder(data).Decode(&o)
 	return o
 }
+
+func (r *OpenDialogRequest) IsValid() error {
+	if r.URL == "" || !IsValidHttpUrl(r.URL) {
+		return errors.New("Invalid URL")
+	}
+
+	if r.TriggerId == "" {
+		return errors.New("Empty trigger id")
+	}
+
+	return r.Dialog.isValid()
+}
+
+func (d *Dialog) isValid() error {
+	if d.Title == "" || len(d.Title) > 24 {
+		return errors.New("invalid dialog title")
+	}
+
+	if d.IconURL != "" && !IsValidHttpUrl(d.IconURL) {
+		return errors.New("invalid icon url")
+	}
+
+	if len(d.Elements) > 5 {
+		return errors.New("The maximum number of dialog elements is 5")
+	}
+
+	if len(d.Elements) != 0 {
+		elementMap := make(map[string]bool)
+
+		for i := 0; i < len(d.Elements); i++ {
+			if elementMap[d.Elements[i].DisplayName] {
+				return errors.New("duplicate dialog element name")
+			}
+			elementMap[d.Elements[i].DisplayName] = true
+
+			err := d.Elements[i].isValid()
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func (d *DialogElement) isValid() error {
+	textSubTypeMap := map[string]bool{
+		"":         true,
+		"text":     true,
+		"email":    true,
+		"number":   true,
+		"tel":      true,
+		"url":      true,
+		"password": true,
+	}
+
+	if d.MinLength < 0 {
+		return errors.New("invalid minLength")
```

Comment:
It's probably best to use the JSON keys in these errors, as these will be sent back via HTTP error. The caller could be written in any language, so JSON seems to be the best representation to refer to in the response.

```suggestion
		return errors.New("min_length cannot be a negative number")
```